compiler/rustc_macros/src/diagnostics/fluent.rs

```diff
@@ -81,217 +38,215 @@ fn invocation_relative_path_to_absolute(span: Span, path: &str) -> PathBuf {
-                    let snake_name = Ident::new(&attr_name.replace('-', "_"), span);
-                    if !previous_attrs.insert(snake_name.clone()) {
-                        continue;
-                    }
-
-                    if attr_name.contains('-') {
-                        Diagnostic::spanned(
-                            path_span,
-                            Level::Error,
-                            format!("attribute `{attr_name}` contains a '-' character"),
-                        )
-                        .help("replace any '-'s with '_'s")
-                        .emit();
-                    }
-
-                    constants.extend(quote! {
-                        pub const #snake_name: crate::SubdiagnosticMessage =
-                            crate::SubdiagnosticMessage::FluentAttr(
-                                std::borrow::Cow::Borrowed(#attr_name)
-                            );
-                    });
-                }
             }
         }
+    }
 
-        if let Err(errs) = bundle.add_resource(resource) {
-            for e in errs {
-                match e {
-                    FluentError::Overriding { kind, id } => {
-                        Diagnostic::spanned(
-                            path_span,
-                            Level::Error,
-                            format!("overrides existing {}: `{}`", kind, id),
-                        )
-                        .span_help(previous_defns[&id], "previously defined in this resource")
-                        .emit();
-                    }
-                    FluentError::ResolverError(_) | FluentError::ParserError(_) => unreachable!(),
+    if let Err(errs) = bundle.add_resource(resource) {
+        for e in errs {
+            match e {
+                FluentError::Overriding { kind, id } => {
+                    Diagnostic::spanned(
+                        resource_span,
+                        Level::Error,
+                        format!("overrides existing {}: `{}`", kind, id),
+                    )
+                    .emit();
                 }
+                FluentError::ResolverError(_) | FluentError::ParserError(_) => unreachable!(),
             }
         }
-
-        includes.extend(quote! { include_str!(#relative_ftl_path), });
-
-        let ident = res.ident;
-        generated.extend(quote! {
-            pub mod #ident {
-                #constants
-            }
-        });
     }
 
+    let crate_name = Ident::new(&crate_name, resource_str.span());
     quote! {
+        pub static DEFAULT_LOCALE_RESOURCE: &'static str = include_str!(#relative_ftl_path);
+
         #[allow(non_upper_case_globals)]
         #[doc(hidden)]
-        pub mod fluent_generated {
-            pub static DEFAULT_LOCALE_RESOURCES: &'static [&'static str] = &[
-                #includes
-            ];
-
-            #generated
+        pub(crate) mod fluent_generated {
+            pub mod #crate_name {
```

Comment:
Why is `mod #crate_name` necessary?

Reply:
This is needed for the `cratename_error_name` -> `cratename::error_name` slug translation. A slug called `parser_incorrect_semicolon` in the ftl file gets referenced as `parser::incorrect_semicolon` in Rust code. In theory it's not *neccessary* because you could strip that prefix at least for crate-local use (on the ftl side it might be a good idea to keep the prefixes, because then they could live in the same namespace). This stripping would require changing all the code that references fluent slugs. That's possible but increases the conflict potential of this PR even further, so it might be better to do that in a later PR.